### PR TITLE
Feat: Locale Support

### PR DESCRIPTION
**Pull request by beausmith**
TODO: Update the version of this module and then update version number in `election-manager`.

### CODE REVIEW

**Review by benadida — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -11,6 +11,17 @@ yarn install
 yarn pnpify --sdk
 ```
 
+## Locally Dev on this Module in another Project
+
+In dev `ballot-encoder` dir, run `yarn tsc` to generate javascript files.
+
+In the PROJECT you wish to use this dev version of `ballot-encoder`:
+
+1. `cd path/to/PROJECT/client/node_modules/@votingworks`
```
Comment (eventualbuddha) [inline README.md]:
```suggestion
1. `cd path/to/PROJECT/node_modules/@votingworks`
```

File: README.md [hunk 2]
```diff
@@ -11,6 +11,17 @@ yarn install
 yarn pnpify --sdk
 ```
 
+## Locally Dev on this Module in another Project
```
Comment (eventualbuddha) [inline README.md]:
```suggestion
## Developing Alongside another Project
```

File: README.md [hunk 3]
```diff
@@ -11,6 +11,17 @@ yarn install
...
+
+In dev `ballot-encoder` dir, run `yarn tsc` to generate javascript files.
```
Comment (eventualbuddha) [inline README.md]:
```suggestion
In dev `ballot-encoder` dir, run `yarn tsc` to generate JavaScript files.
```

File: README.md [hunk 4]
```diff
@@ -11,6 +11,17 @@ yarn install
...
+
+In the PROJECT you wish to use this dev version of `ballot-encoder`:
+
+1. `cd path/to/PROJECT/client/node_modules/@votingworks`
+2. `rm -rf ballot-encoder`
+3. `ln -s path/to/ballot-encoder ballot-encoder`
+4. restart project server
```
Comment (eventualbuddha) [inline README.md]:
```suggestion
4. restart project server if applicable
```

File: src/election.ts [hunk 5]
```diff
@@ -202,11 +246,14 @@ export const validateVotes = ({
...
 }
 
-export const electionSample = electionSampleUntyped as Election
+export const electionSample = (electionSampleUntyped as unknown) as LocalizedElection
+export const electionSampleLongContent = (electionSampleLongContentUntyped as unknown) as LocalizedElection
 
 /**
  * Gets the adjective used to describe the political party for a primary
  * election, e.g. "Republican" or "Democratic".
+ * TODO: Deprecate this function. It does not support i18n and 'party.fullname`
```
Comment (eventualbuddha) [inline src/election.ts]: You can use `@deprecated`.

File: src/election.ts [hunk 6]
```diff
@@ -222,6 +269,22 @@ export const getPartyPrimaryAdjectiveFromBallotStyle = ({
...
+export const getPartyFullNameFromBallotStyle = ({
+  ballotStyleId,
+  election,
+}: {
+  ballotStyleId: string
+  election: Election
+}): string => {
+  const ballotStyle = getBallotStyle({ ballotStyleId, election })
+  const party = election.parties.find((p) => p.id === ballotStyle?.partyId)
+  return party?.fullName || ''
```
Comment (eventualbuddha) [inline src/election.ts]:
```suggestion
  return party?.fullName ?? ''
```

File: src/election.ts [hunk 7]
```diff
@@ -290,3 +353,72 @@ export function vote(
...
 }
+
+/**
+ * Helper function to get array of locale codes used in election definition.
+ */
+export const getElectionLocales = (
+  election: Election,
+  baseLocale = 'en-US'
+): string[] =>
+  [baseLocale].concat(
```
Comment (eventualbuddha) [inline src/election.ts]: I like `[baseLocale, ...Object.keys(`, but your call.

**Review by eventualbuddha — APPROVED** [review]
Seems good, just a few nits if you want to address them.

**Pull request merged**